homeassistant/components/min_max/sensor.py

```diff
@@ -110,6 +113,26 @@ def calc_mean(sensor_values, round_digits):
     return round(sensor_value_sum / count, round_digits)
 
 
+def calc_median(sensor_values, round_digits):
+    """Calculate median value, honoring unknown states."""
+    count = 0
+    sensor_value_list = []
+    for _, sensor_value in sensor_values:
+        if sensor_value != STATE_UNKNOWN:
+            count += 1
+            sensor_value_list.append(sensor_value)
+    if count == 0:
+        return None
+    sensor_value_list.sort()
+    if count % 2 == 0:
+        median1 = sensor_value_list[count // 2]
+        median2 = sensor_value_list[count // 2 - 1]
```

Comment:
Since we're rounding down, shouldn't this be +1 ?

Reply:
even though `median1` is a floor division of the list length (rounded down), it is used to get element x from the list which starts with first index being 0, hence the element is the x + 1 element in the list. Could probably been explained better, but it is mathematically correct.